### PR TITLE
Resolve breaking change of adding a method to ChangeNotifier.

### PR DESCRIPTION
**Pull request by polina-c**
Details: https://github.com/flutter/flutter/pull/133060#discussion_r1327979244

### CODE REVIEW

File: packages/flutter/lib/src/foundation/change_notifier.dart [hunk 1]
```diff
@@ -227,16 +227,16 @@ mixin class ChangeNotifier implements Listenable {
   /// Make sure to invoke it with condition `if (kFlutterMemoryAllocationsEnabled) ...`
   /// so that the method is tree-shaken away when the flag is false.
   @protected
-  void maybeDispatchObjectCreation() {
+  static void maybeDispatchObjectCreation(ChangeNotifier object) {
```
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/change_notifier.dart:230]: nit: update the doc above to document the parameter.

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by goderbauer** [comment]
@polina-c Did the previous change already make it into a release? If so, we likely want to cherry-pick this change into that release.

**Comment by passsy** [comment]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

**Comment by polina-c** [comment]
Google testing passed: cl/566364786

**Pull request merged**